### PR TITLE
Add an HSI attribute for Intel SMAP

### PR DESCRIPTION
**Pull request by hughsie**
See https://en.wikipedia.org/wiki/Supervisor_Mode_Access_Prevention for details.

### CODE REVIEW

File: plugins/cpu/fu-plugin-cpu.c [hunk 1]
```diff
@@ -46,6 +47,9 @@ fu_plugin_coldplug (FuPlugin *plugin, GError **error)
 			data->has_cet = TRUE;
 		if (fu_cpu_device_has_flag (dev, FU_CPU_DEVICE_FLAG_TME))
 			data->has_tme = TRUE;
+		if (fu_cpu_device_has_flag (dev, FU_CPU_DEVICE_FLAG_SMEP) &&
+		    fu_cpu_device_has_flag (dev, FU_CPU_DEVICE_FLAG_SMAP))
+			data->has_smap = TRUE;
```
Comment (superm1) [inline plugins/cpu/fu-plugin-cpu.c:52]: At least the way the wikipedia page reads, SMAP is the extension to SMEP.  I think you just want to look for SMAP.

Comment (hughsie) [reply plugins/cpu/fu-plugin-cpu.c:52]: So maybe just ignore SMEP completely?

Comment (superm1) [reply plugins/cpu/fu-plugin-cpu.c:52]: Assuming wikipedia is right, yeah I think so.

Comment (hughsie) [reply plugins/cpu/fu-plugin-cpu.c:51]: Gotcha, fixed.

**Pull request merged**